bin/arrange_data.R

```diff
@@ -13,148 +18,232 @@ option_list = list(
+                metadata[i_m, extra_attributes$tag[j] ] <- extra_attributes$value[j]
+            }
+        }
+    
+    } else {
+        extra_attributes <- fromJSON( samples$extra_attributes[i] )
+        extra_attributes$tag <- gsub("[()]", "", gsub(" ", "_", extra_attributes$tag))
+        # add values
+        for (j in 1:length( extra_attributes$tag ) ){
+            if ( extra_attributes$tag[j] %in% names(metadata) ) {
+                metadata[i_m, extra_attributes$tag[j] ] <- extra_attributes$value[j]
+            }
+        }
+    }
+}
 
-    # Now check the file is actually available for download
 
-    if (! is.na(opt$dbox_listing)){
-        ega_files <- readLines(opt$dbox_listing)
-    }else{
-        ega_file_info <- read.delim(opt$ega_listing)
-        ega_files <- ega_file_info$File_name
-    }
+# 4 update metadata with information from study_experiment_run_sample.csv
+study_experiment_run_sample <-  read.csv( paste0(opt$metadata_dir,'/study_experiment_run_sample.csv') )
 
-    # Assume that the dbox entry contains the file name and the run ID
+matched_indices <- match(metadata$sample_alias, study_experiment_run_sample$sample_alias)
 
-    matches <- apply(sample_info, 1, function(x) which(grepl(x['ega_run_id'], ega_files) & grepl(x['filename'], ega_files)))
-    names(matches) <- sample_info$filename
-    lengths <- unlist(lapply(matches, length))
+metadata$biosample_id <- study_experiment_run_sample$biosample_id[matched_indices]
+metadata$ega_run_id <- study_experiment_run_sample$run_accession_id[matched_indices]
+metadata$ega_experiment_id <- study_experiment_run_sample$experiment_accession_id[matched_indices]
 
-    if (any(lengths != 1)){
-        missing <- names(lengths)[lengths == 0]
-        if (length(missing) > 0){
-            write(paste("can't find the following in the downloads:", paste(missing, collapse=', ')), stderr())
-        }
-        multiple <- names(lengths)[lengths > 1]
-        if (length(multiple) > 0){
-            write(paste("The following match multiple files in downloads:", paste(multiple, collapse=', ')), stderr())
-        }
+metadata$library_strategy <- study_experiment_run_sample$library_strategy[matched_indices]
+metadata$library_source <- study_experiment_run_sample$library_source[matched_indices]
+metadata$library_layout <- study_experiment_run_sample$library_layout[matched_indices]
 
-        q(status=1)
-    }
+metadata$library_name <- study_experiment_run_sample$library_name[matched_indices]
+metadata$library_selection <- study_experiment_run_sample$library_selection[matched_indices]
+metadata$study_title <- study_experiment_run_sample$study_title[matched_indices]
+metadata$study_type <- study_experiment_run_sample$study_type[matched_indices]
+metadata$instrument_platform <- study_experiment_run_sample$instrument_platform[matched_indices]
+metadata$instrument_model <- study_experiment_run_sample$instrument_model[matched_indices]
```

Comment:
It can't be used for all the chunk, as source and dest columns names are not equal, e.g. 
```
metadata$ega_run_id <- study_experiment_run_sample$run_accession_id[matched_indices]
metadata$ega_experiment_id <- study_experiment_run_sample$experiment_accession_id[matched_indices]
```
but I'll apply for the rest

Reply:
[45b4c98](https://github.com/ebi-gene-expression-group/ega_downloader/pull/9/commits/45b4c987f7b1d1bef8dd7d3b2c90459768c7cdf0)

Reply:
> It can't be used for all the chunk, as source and dest columns names are not equal, e.g.
> 
> ```
> metadata$ega_run_id <- study_experiment_run_sample$run_accession_id[matched_indices]
> metadata$ega_experiment_id <- study_experiment_run_sample$experiment_accession_id[matched_indices]
> ```
> 
> but I'll apply for the rest

Thats where it uses `columns` to go through all variables.. But happy with your solution as it minimises manual changes in future